robotis_op3:
  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 50

# trajectory Controllers ---------------------------------------              
  leg_traj_controller:
    type: effort_controllers/JointTrajectoryController
    joints:
      - l_hip_yaw
      - l_hip_roll
      - l_hip_pitch
      - l_knee
      - l_ank_roll
      - l_ank_pitch
      - r_hip_yaw
      - r_hip_roll
      - r_hip_pitch
      - r_knee
      - r_ank_roll
      - r_ank_pitch
    gains:
      l_hip_yaw:    {p: 50.0, i: 0.0, d: 0.0, i_clamp: 1}
      l_hip_roll:   {p: 50.0, i: 0.0, d: 0.0, i_clamp: 1}
      l_hip_pitch:  {p: 50.0, i: 0.0, d: 0.0, i_clamp: 1}
      l_knee:       {p: 50.0, i: 0.0, d: 0.0, i_clamp: 1}
      l_ank_roll:   {p: 50.0, i: 0.0, d: 0.0, i_clamp: 1}
      l_ank_pitch:  {p: 50.0, i: 0.0, d: 0.0, i_clamp: 1}
      r_hip_yaw:    {p: 50.0, i: 0.0, d: 0.0, i_clamp: 1}
      r_hip_roll:   {p: 50.0, i: 0.0, d: 0.0, i_clamp: 1}
      r_hip_pitch:  {p: 50.0, i: 0.0, d: 0.0, i_clamp: 1}
      r_knee:       {p: 50.0, i: 0.0, d: 0.0, i_clamp: 1}
      r_ank_roll:   {p: 50.0, i: 0.0, d: 0.0, i_clamp: 1}
      r_ank_pitch:  {p: 50.0, i: 0.0, d: 0.0, i_clamp: 1}

  arm_traj_controller:
    type: effort_controllers/JointTrajectoryController
    joints:
      - l_sho_pitch
      - l_sho_roll
      - l_el
      - r_sho_pitch
      - r_sho_roll
      - r_el
    gains:
      l_sho_pitch: {p: 10, i: 0, d: 0.0}
      l_sho_roll: {p: 10, i: 0, d: 0.0}
      l_el: {p: 10, i: 0, d: 0.0}
      r_sho_pitch: {p: 10, i: 0, d: 0.0}
      r_sho_roll: {p: 10, i: 0, d: 0.0}
      r_el: {p: 10, i: 0, d: 0.0}

  head_pan_position:
    type: effort_controllers/JointPositionController
    joint: head_pan
    pid: {p: 5.0, i: 0.0, d: 0.0}
  head_tilt_position:
    type: effort_controllers/JointPositionController
    joint: head_tilt
    pid: {p: 5.0, i: 0.0, d: 0.0}
